src/views/SourcesView.vue

```diff
@@ -1,250 +1,187 @@
+});
 
 /**
  * Get all available sources
  */
 async function fetchAllStreams() {
-  isLoadingScreensAndWindows.value = true
-  await getVideoSources()
-  console.log(desktopCaptureSources.value)
-  await loadStreamSources()
-  isLoadingScreensAndWindows.value = false
-}
-
-/**
- * Get all the video sources available on the system
- */
-async function getVideoSources() {
-  // TODO: is not called again if new sources showes up
-  // Get the available video sources
-  desktopCaptureSources.value = await (window as any).electronAPI.getMedia()
+    // wait for the fetched screens and windows
+    desktopVideoStreamSources.value =
+        await streamHandler.getScreenAndApplicationSources();
+
+    // Loop through all sources and set the MediaStream to video nodes
+    Array.from(desktopVideoStreamSources.value).forEach(function (element) {
+        setSourceForVideoNode(element);
+    });
 }
 
-/**
- * Load the proper media stream source for the video element
- */
-async function loadStreamSources() {
-  // Is there a better way then this?
-  const videoElements: HTMLVideoElement | null =
-    document.querySelectorAll('.preview')
-
-  Array.from(videoElements).forEach(function (video, index) {
-    setSourceForVideoNode(desktopCaptureSources.value[index], video)
-  })
-}
-
-/**
- * Get the stream source
- * @param {string} sourceName
- */
-async function getStreamSource(source: any) {
-  const constraints: any = {
-    video: {
-      mandatory: {
-        chromeMediaSource: 'desktop',
-        chromeMediaSourceId: source.id,
-      },
-    },
-  }
-
-  // Create a Stream
-  const stream = await navigator.mediaDevices.getUserMedia(constraints)
-
-  return stream
-}
-
-/**
- * Load the stream sources
- * @param {any} source
- * @param {HTMLVideoElement} videoHTMLNode
- * @returns {Promise<void>}
- */
-async function setSourceForVideoNode(source: any, videoNode: HTMLVideoElement) {
-  const constraints: any = {
-    video: {
-      mandatory: {
-        chromeMediaSource: 'desktop',
-        chromeMediaSourceId: source.id,
-      },
-    },
-  }
-
-  // Create a Stream
-  const stream = await navigator.mediaDevices.getUserMedia(constraints)
-
-  // Preview the source in a video element
-  videoNode.srcObject = stream
-}
-
-/**
- * Load a specific stream source as main video
- *
- * @param {any} source
- */
-async function selectSource(source: any) {
-  // Set the selected source
-  selectedSource.value = source
-
-  // Preview the source in a video element
-  const videoElement: HTMLVideoElement | null =
-    document.querySelector('#mainVideo')
-
-  setSourceForVideoNode(source, videoElement)
+async function setSourceForVideoNode(source: any) {
+    const constraints: any = {
+        video: {
+            mandatory: {
+                chromeMediaSource: 'desktop',
+                chromeMediaSourceId: source.id,
+            },
+        },
+    };
+
+    streams.value.push(await navigator.mediaDevices.getUserMedia(constraints));
 }
 </script>
```

Comment:
seems to me like almost too much typescript (logic in general) for a view, but not sure. Some of this outsourceable to proc/? Imo regarding logic, views should only contain typescript for - only - receiving (onclick-) events and actual DOM changes.

Reply:
I know that the last function "setSourceForVideoNode" is because if i added this ot the StreamHandler for some reason the videos could properly load, that's why i put it there again.

Reply:
Okay. I mean it's still acceptable but generally and for the future we should have as less ts as possible in these files. Can be resolved from my side